Source/engine.h

```diff
@@ -226,8 +226,7 @@ struct Size {
 
 struct Rectangle {
 	Point Position;
-	int w;
-	int h;
+	Size Size;
 };

```

Comment:
I don't think building the rectangle struct out of a `Point` and `Size` is ideal. x, y, width and height seam inherent to a rectangle. It doesn't seam to benefit any specific areas of the code either, but if it would help I think a method for getting the origin point from a rectangle might make sens instead.

Reply:
I'd disagree. I think building on top of `Point` and `Size` makes the most sense by far here. We could, optionally, expose other properties of the rectangle as we see fit, but I didn't feel the need to do so at this point.

"x, y" is a `Point`, and "width and height" is a `Size`. There is no sensible reason to downgrade the abstraction and use their properties directly in a rectangle.

This is vaguelly (not that vaguelly) based on `System.Drawing.Rectangle` from here:
https://docs.microsoft.com/en-us/dotnet/api/system.drawing.rectangle?view=net-5.0

Reply:
I like the idea of composing a rect from those properties (instead of using multiple inheritance) but it does make it a bit verbose when trying to access specific dimensions of a Rectangle. Using .Net as an example, `x` and `y` properties exist as aliases for `Location.x` and `Location.y` respectively. Having accessors available in devilutionx::Rectangle would be nice as a shorter way to refer to these values.

Something like the following?
```cpp
constexpr int &x() { return Position.x; }
constexpr int &y() { return Position.y; }
constexpr int &height() { return Size.height; }
constexpr int &width() { return Size.width; }
```

Reply:
I thought about creating these but didn't feel an immediate need to. I'll add them later today.

Reply:
This PR ended up being about using rectangle in a few places (instead of adding it) as it was added in a different PR. Will send a separate PR later to add these shortcuts.